presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java

```diff
@@ -24,6 +24,11 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+/**
+ * Represents a piece of a file at a given path starting at a given position and continuing for length bytes.
+ * All numeric values must be zero or greater. Object values must be non-null.
+ * There is no guarantee the file exists, or, if it exists, that it is large enough to provide length bytes.
```

Comment:
I would remove this as well. I guess technically, there's no guarantee about these things because this is just a data structure storing some information, but we do get the path from the metastore and then get metadata for the file from the file system to be able to populate this, so if we exclude concurrent updates, then we do have guarantees about this.

Reply:
The API doc for a public class should not assume the class will only be used in a particular way. Unless we can make this class non-public (which maybe we can do? I've noticed Presto tends to have more public API than it needs.) This class provides no such guarantees.

Reply:
maybe rephrase it to say that the creator of a HiveFileSplit is responsible for ensuring the existence of the file, start and length, and this class does not do any such validation itself.

Reply:
rephrased